specific/cmo_stock/models/stock.py

```diff
@@ -0,0 +1,132 @@
+                         'search_default_group_by_product': True}"
+        }
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    project_id = fields.Many2one(
+        'project.project',
+        string='Project name',
+        domain=lambda self: self._get_domain(),
+    )
+    location_dest_id = fields.Many2one(
+        'stock.location',
+        default=lambda self: self._get_location_dest_id(),
+    )
+
+    @api.model
+    def _get_domain(self):
+        user = self.env['res.users'].browse(self._uid)
+        operating_unit_ids = []
+        for operating_unit in user.operating_unit_ids:
+            operating_unit_ids.append(operating_unit.id)
+        return [('operating_unit_id', 'in', operating_unit_ids)]
+
+    @api.model
+    def _get_location_dest_id(self):
+        Location = self.env['stock.location']
+        user = self.env['res.users'].browse(self._uid)
+        ou_id = user.default_operating_unit_id and \
+            user.default_operating_unit_id.id or False
+        if ou_id:
+            location = Location.search([('operating_unit_id', '=', ou_id)])
+        return location and location[0].id or False
+
+    # @api.multi
+    # @api.onchange('product_id', 'location_id', 'location_dest_id',
+    #               'picking_id.partner_id')
+    # def onchange_product_id(self):
+    #     for move in self:
+    #         res = super(StockMove, move).onchange_product_id(
+    #             move.product_id.id, move.location_id.id,
+    #             move.location_dest_id.id, move.picking_id.partner_id.id
+    #         )
+    #         if res:
+    #             res = res['value']
+    #             move.name = res.get('name', False)
+    #             move.product_uom = res.get('product_uom', False)
+    #             move.product_uos = res.get('product_uos', False)
+    #             move.product_uom_qty = res.get('product_uom_qty', False)
+    #             move.product_uos_qty = res.get('product_uos_qty', False)
+    #             move.location_id = res.get('location_id', False)
+    #             move.location_dest_id = res.get('location_dest_id', False)
+
+
+class StockPickingType(models.Model):
+    _inherit = 'stock.picking.type'
+
+    @api.multi
+    def action_picking_type_form(self):
+        user = self.env['res.users'].browse(self._uid)
+        domain = [(1, '=', 1)]
+        if user.has_group('stock.group_stock_manager'):
+            domain = ['|', ('code', '=', 'incoming'),
+                      '|', ('code', '=', 'outgoing'),
+                           ('code', '=', 'internal')]
+        elif user.has_group('cmo_stock.group_stock_wh_user'):
+            domain = ['|', ('code', '=', 'incoming'),
+                           ('code', '=', 'outgoing')]
+        elif user.has_group('stock.group_stock_user'):
+            domain = [('code', '=', 'outgoing')]
+        elif user.has_group('cmo_stock.group_stock_readonly'):
+            domain = [('code', '=', 'outgoing')]
+        return {
```

Comment:
Can we reuse the action?
(Please come to me if you don't understand what I mean)